### ISSUE TITLE
Output initialization problem

### ISSUE THREAD

**Issue by aaronchch** [dotnet/iot#721]
I have a Raspberry Pi 3B and I'm using an output to control a relay board. The relay board requires and active-low signal
Using the code below to initialize the output causes it to go low for long enough for the relay to briefly close before the output goes high. 

```
GpioController controller = new GpioController(PinNumberingScheme.Logical);
controller.OpenPin(18, PinMode.Output);
controller.Write(18, PinValue.High);
```

What is needed is a way of setting the output state of the pin before configuring it as an output. Calling GpioController.Write before opening the pin throws an exception.

**Comment by Frankenslag** [comment]
@aaronchch 

How about

```
GpioController controller = new GpioController(PinNumberingScheme.Logical);
controller.OpenPin(18);
controller.Write(18, PinValue.High);
controller.SetPin Mode(18, PinPinMode.Output);
```

Or perhaps....

```
GpioController controller = new GpioController(PinNumberingScheme.Logical);
controller.OpenPin(18, PinMode.InputPullUp);
controller.Write(18, PinValue.High);
controller.SetPin Mode(18, PinPinMode.Output);
```

Note that I haven't tried any of this and I don't know if it is safe to do or even if it will work...

**Comment by krwq** [comment]
@aaronchch interesting scenario - I think we should add this API assuming this is supported by the OS. For now for Raspberry PI on Linux you can also access registers directly using:
https://github.com/dotnet/iot/blob/master/src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs#L313

note they are protected so you will need to inherit from that class, expose those 2 APIs and pass in the instance to GpioController.

What about something like: `controller.OpenPin(18, PinMode.Output, PinValue.High;`

**Comment by joperezr** [comment]
I think that even if we get the initial value on the OpenPin method, we might have problems honoring this in some drivers (like the unix driver) but we can try do our best to set it just as the pin gets opened. I have triaged this in for our vNext as I think it would make sense to add support for this.

**Comment by krwq** [comment]
@joperezr I think we should just document API as "best effort but no guarantee" (similarly as CancellationToken for Async APIs) - we can also add full list of drivers which would support this

**Comment by aaronchch** [comment]
Sounds perfect and thanks for the workaround.

**Comment by vicente-da-silva** [comment]
Any feedback on when this will be implemented? Having the relays bounce on application startup is not good.

**Comment by pgrawehr** [comment]
There's a PR open to potentially fix this: #1361. We'll be discussing this on our next meeting.

**Comment by krwq** [comment]
Triage: API `controller.OpenPin(18, PinMode.Output, PinValue.High);` is approved as long as we find at least one driver which supports it. And we should fallback to the Open+Write behavior if it's not possible to achieve with this driver. If we can't find any driver which can do this as atomic operation we should not add this API.
